60_network_learning.Rmd

```diff
@@ -0,0 +1,654 @@
+centr_df$eigenvec <- igraph::eigen_centrality(spring_graph)$vector
+
+rownames(centr_df) <- rownames(spring_net$adjaMat1)
+head(centr_df, 15)
+```
+
+#### Scale node sizes by degree
+
+Centrality measures can be visualized in the network plot by scaling the node sizes according to one of these measures. We plot the four networks based on SparCC, Proportionality, SpiecEasi, and SPRING once more, but this time we scale the node sizes by their degrees. The layout is the same as before. The `igraph` plot function is again used for the visualization.
+
+```{r netplot_degree, fig.width=7, fig.height=7, out.width='80%'}
+deglist <- lapply(graphlist, igraph::degree)
+```
+
+```{r, fig.height=10, fig.width=10}
+par(mfrow = c(2,2))
+for (i in seq_along(graphlist)) {
+  plot(graphlist[[i]], layout = lay_fr, vertex.size = deglist[[i]], 
+     vertex.label = NA, main = names(graphlist)[i])
+}
+```
+
+#### Degree distribution
+
+The degree distribution is another popular measure that expresses the probability distribution of degrees over the entire network. It thus provides insight into the overall network structure. We plot the degree distribution for all four association estimation methods to compare the network structure.
+
+```{r, message=FALSE, warning=FALSE}
+library(ggplot2)
+```
+
+```{r}
+# Compute degree distributions
+ddlist <- lapply(graphlist, igraph::degree.distribution)
+
+# Maximum degree
+maxdeg <- max(lengths(ddlist))
+
+# Make list elements the same length
+for(i in seq_along(graphlist)) {
+  length(ddlist[[i]]) <- maxdeg
+}
+
+# Data frame needed for ggplot2
+df <- data.frame(Degree = rep(1:maxdeg, length(graphlist)), 
+                 Fraction = unlist(ddlist), 
+                 Method = rep(names(graphlist), each = maxdeg))
+
+ggplot(df, aes(x = Degree, y = Fraction, group = Method)) +
+  geom_line(aes(color = Method)) +
+  geom_point(aes(color = Method)) +
+  theme_bw()
+```
+
+The SparCC and shrinkage proportionality networks have a considerably higher proportion of singletons (zero-degree nodes) than the two conditional dependency graphs, but a lower proportion of nodes with degrees between one and five. The MB and SPRING graphs, on the other hand, have a higher proportion of low degree nodes, but no highly connected nodes with a degree greater than eleven.
+
+#### Clustered heatmaps
+
+Using the `ComplexHeatmap` package, we plot heatmaps of the association matrices for the four considered association measures. Rows and columns are sorted according to the clusters identified via hierarchical clustering.
+
+```{r, message=FALSE, warning=FALSE}
+library(ComplexHeatmap)
+library(circlize)
+library(patchwork)
+```
+
+For each association measure, we select the 50 nodes with the highest sum of edge weights.
+
+```{r select_nodes}
+# Function for selecting taxa with highest sum of edge weights
+select_taxa <- function(adja, ntaxa = 50) {
+  sel <- names(sort(rowSums(adja), decreasing = TRUE))[1:ntaxa]
```

Comment:
use `seq_len(ntaxa)` instead of `1:ntaxa` (current Bioc requirement & best practice)